index.bs

```diff
@@ -2222,6 +2266,99 @@ To <dfn>parse trigger specs</dfn> given a [=map=] |map|, a [=moment=]
+        1. [=set/Append=] |value| to |result|.
+1. If |result| is not [=set/is empty|empty=]:
+    1. If |attributionScopeLimit| is null, return an error.
+1. If |result|'s [=set/size=] is greater than |attributionScopeLimit|, return an error.
+1. If |result|'s [=set/size=] is greater than [=max number of attribution scopes per source=], return an error.
+1. Return |result|.
+
+To <dfn>find sources with common destinations and reporting origin</dfn> given an [=attribution source=] |pendingSource|:
+1. Let |cache| be the user agent's [=attribution source cache=].
+1. Let |matchingSources| be a new [=list=].
+1. [=set/iterate|For each=] |source| of |cache|:
+    1. Let |commonDestinations| be the [=set/intersection=] of |source|'s [=attribution source/attribution destinations=] and |pendingSource|'s [=attribution source/attribution destinations=].
+    1. If |commonDestinations| [=set/is empty=], [=iteration/continue=].
+    1. If |source|'s [=attribution source/reporting origin=] and |pendingSource|'s [=attribution source/reporting origin=] are not [=same origin=], [=iteration/continue=].
+    1. [=list/Append=] |source| to |matchingSources|.
+1. Return |matchingSources|.
+
+To <dfn>Remove event reports for source</dfn> given an [=attribution source=] |source| and a [=moment=] |minTriggerTime|:
+1. Let |matchingReports| be a new [=list=] whose elements are all the elements in the [=event-level report cache=] whose [=event-level report/trigger time=] is greater than or equal to |minTriggerTime| and [=event-level report/event Id=] and |source|'s [=attribution source/event ID=].
+1. [=set/Remove=] |matchingReports| from the [=event-level report cache=].
+
+To <dfn>remove sources with different max number of event states if necessary</dfn> given an [=attribution source=] |pendingSource|:
+1. If |pendingSource|'s [=attribution source/max number of event states=] is null, return.
+1. [=list/iterate|For each=] |source| of [=find sources with common destinations and reporting origin=] with |pendingSource|:
+    1. If |source|'s [=attribution source/max number of event states=] is null or equals |pendingSource|'s [=attribution source/max number of event states=], [=iteration/continue=].
+    1. [=Remove event reports for source=] with |source| and |pendingSource|'s [=attribution source/source time=].
+    1. [=set/Remove=] |source| from the [=attribution source cache=].
+
+To <dfn>remove sources with lower attribution scope limit from cache if necessary</dfn> given an [=attribution source=] |pendingSource|:
+1. If |pendingSource|'s [=attribution source/attribution scope limit=] is null, return.
+1. [=list/iterate|For each=] |source| of [=find sources with common destinations and reporting origin=] with |pendingSource|:
+    1. If |source|'s [=attribution source/attribution scope limit=] is null or greater than or equal to |pendingSource|'s [=attribution source/attribution scope limit=], [=iteration/continue=].
```

Comment:
Is there a reason we don't remove fake reports for this algorithm?

Reply:
Nice catch, done.